### PR TITLE
2018.3 Upgrade

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Upgraded the GDK project to 2018.3.2f1 (latest release).

- Removed incremental compiler as a direct dependency
- Upgraded entities to `preview-21` (latest is `preview-22`, but has compile errors)
- Added a safety check for the workaround for crashes on Linux workers. 
- Some Unity asset changes.
- Ran linting on a file which had some wacky formatting.

Left to do:

- [x] Change editor hash and such for CI.
- [x] Update test project

Another step on the road to #676 

#### Tests
Ran local deployment, built workers for cloud, ran 

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

Will need to change the setup guide docs.

**Did you remember a changelog entry?**
Yeeee

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,9 @@
 - `Improbable.Gdk.EntityTemplate` is now mutable and exposes a set of APIs to add, remove, and replace component snapshots
     - This replaces the `Improbable.Gdk.Core.EntityBuilder` class.
     - These changes also allow you to reuse an `EntityTemplate` more than once.
+- Upgraded the project to be compatible `2018.3.2f1`.
```
Comment (paulbalaji) [inline CHANGELOG.md]: "...compatible _with_..."

Comment (jessicafalk) [inline CHANGELOG.md]: to be compatible **with**

Comment (jamiebrynes7) [reply CHANGELOG.md]: Nah

File: workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs [hunk 2]
```diff
@@ -110,69 +116,106 @@ private Worker(string workerType, Connection connection, ILogDispatcher logDispa
...
         public static async Task<Worker> CreateWorkerAsync(
             LocatorConfig parameters,
             ConnectionParameters connectionParameters,
             ILogDispatcher logger, Vector3 origin)
         {
-                using (var locator = new Locator(parameters.LocatorHost, parameters.LocatorParameters))
+            // TODO: Remove when UTY-1578 is fixed.
+#if UNITY_STANDALONE_LINUX
+            connectionParameters.EnableProtocolLoggingAtStartup = false;
+            Debug.LogWarning("Automatically disabling protocol logging on Linux workers to prevent crashes.");
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs]: should we always print that warning? or only if we actually do disable it

Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs]: I actually think it doesn't matter either way in this particular case

**Pull request merged**